src/Integration/NewConnectedMode/ObsoleteConfigurationProvider.cs

```diff
@@ -51,12 +51,11 @@ internal class ObsoleteConfigurationProvider : IObsoleteConfigurationProviderSer
         [ImportingConstructor]
         public ObsoleteConfigurationProvider(
             [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
-            ICredentialStoreService credentialStoreService,
-            ILogger logger)
+            ISolutionBindingDataReader solutionBindingDataReader)
             : this(
                 new LegacySolutionBindingPathProvider(serviceProvider),
                 new ObsoleteConnectedModeSolutionBindingPathProvider(serviceProvider),
-                new SolutionBindingDataReader(new SolutionBindingFileLoader(logger), new SolutionBindingCredentialsLoader(credentialStoreService)))
```

Comment:
This is the core of the change - made the data reader MEF-exportable so it is simpler to construct, since we'll need one for the new Connected Mode Config provider too.
All of the other changes follow from fixing up the constructors.